libraries/chain/include/graphene/chain/asset_object.hpp

```diff
@@ -164,6 +165,44 @@ namespace graphene { namespace chain {
          template<class DB>
          share_type reserved( const DB& db )const
          { return options.max_supply - dynamic_data(db).current_supply; }
+
+         /// @return true if asset can accumulate fees in the given denomination
+         template<class DB>
+         bool can_accumulate_fee(const DB& db, const asset& fee) const {
+            return (( fee.asset_id == get_id() ) ||
+                    ( is_market_issued() && fee.asset_id == bitasset_data(db).options.short_backing_asset ));
+         }
+
+         /***
+          * @brief receive a fee asset to accrue in dynamic_data object
+          *
+          * Asset owners define various fees (market fees, force-settle fees, etc.) to be
+          * collected for the asset owners. These fees are typically denominated in the asset
+          * itself, but for bitassets some of the fees are denominated in the collateral
+          * asset. This will place the fee in the right container.
+          */
+         template<class DB>
+         void accumulate_fee(DB& db, const asset& fee) const
+         {
```

Comment:
Please add a `FC_ASSERT` for `fee.amount > 0` here as a defensive measure.

Reply:
Should this be an FC_ASSERT or a quiet return (no-op)?

Seems to me the responsibility to not enter this code branch if fee-value is zero lies higher up the call stack, meaning a check here would be redundant.  And if a fee-calculation did result in a zero fee-value, (e.g. round-down on a low fee percent on a dust amount), and if the calling function omitted "if value > 0", then the expected behavior here would be to quietly "apply" the non-change, as this is not actually an exceptional condition at the calling level.

Adding a quiet return would avoid an unnecessary call to db.modify though, so I do see that as a good idea.

Reply:
Wait — I had thought share_type was a uint — but I see it's based on int64.  Will add FC_ASSERT to ensure non-negative.  But I think case where amount is exactly zero should be a quiet return.

Reply:
I agree that it's good to add an assertion for non-negative amount and a return for zero amount.

Reply:
Done in 02f9da8.